analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.CSharp8.cs

```diff
@@ -51,13 +50,12 @@ int SwitchExpression_Results()
             }
 
             a = false switch { true => true, _ => false };      // Noncompliant:    true branch is always false
-                                                                // Noncompliant@-1: default branch is always true
             if (a)                                              // Noncompliant
             {
                 return 42;
             }
 
-            a = 0 switch { 1 => true, _ => false };             // Noncompliant
+            a = 0 switch { 1 => true, _ => false };             // Compliant, we don't raise in default statement
```

Comment:
This should raise on the `1` arm, though? Please investigate what's happening here.

Reply:
It's an FN - it seems like it does not learn some constrain about the value of a.
Once it's bool it raises.